libs/StateSync-cpp/include/Microsoft/MixedReality/Sharing/StateSync/NetworkManager.h

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+#include <Microsoft/MixedReality/Sharing/Common/RefPtr.h>
+#include <Microsoft/MixedReality/Sharing/Common/VirtualRefCountedBase.h>
+
+#include <memory>
+#include <string>
+
+namespace Microsoft::MixedReality::Sharing::StateSync {
+
+class NetworkConnection;
+class NetworkListener;
+
+class NetworkManager : public VirtualRefCountedBase {
+ public:
+  // Returns a connection object that can be used to send messages to the remote
+  // endpoint described by connection_string
+  virtual RefPtr<NetworkConnection> GetConnection(
```

Comment:
nit: Is this GetOrCreate? Or OpenConnection?

Reply:
Unspecified for now.
We can discuss the ideal shape later when we are less time-pressured.

Currently it's a thing that can be used to send messages. What it does under the hood (connects immediately or on demand, etc.), and how you can observe its state, is not very relevant for rolling out the prototype. The demo implementation, for example, won't have to connect at all.